### DISCUSSION TITLE
Run dependent job even on failure?

### DISCUSSION THREAD

**Discussion by nirvdrum** [community/community#25388] [Actions]
<p>Hi,</p>
<p>I’m looking to split up a CD process into multiple jobs, some of which can run in parallel, and then aggregate the results in a final job. I have a dependency graph that looks like:</p>
<pre><code class="lang-auto">     A
   /   \
  B     C
   \   /
     D
</code></pre>
<p><strong>A</strong> creates a <em>deployment</em> using the GitHub API. <strong>B</strong> and <strong>C</strong> are jobs run in parallel related to deploying the application. And <strong>D</strong> is a status aggregator that should change the state of the whole deployment from <em>pending</em> to <em>success</em> or <em>failure</em>. I don’t want <strong>B</strong> or <strong>C</strong> to take whatever action on failure because it may be duplicated. In the case of updating a GitHub deployment, that’s not a huge deal. For other actions, however, it would be problematic.</p>
<p>Since deployments only happen when certain branches or tags are updated, <strong>A</strong> has an <em>if</em> guard on <code>github.ref</code>. I don’t need the guards elsewhere because the dependency hierarchy is established and the the whole job graph will show as skipped if the ref doesn’t match. That part works very nicely.</p>
<p>The problem I’m having is getting <strong>D</strong> to work the same way. I want to run it even if <strong>B</strong> or <strong>C</strong> fail, but I only want to run it when <strong>A</strong> runs. I can’t find a way to ignore the failure status of its parents other than using <code>if: always()</code>. I’ve tried to use a different gate expression (e.g., <code>if: ${{ needs.A.outputs.deployment_id }}</code>), but if I use anything other than <code>if: always()</code>, the job won’t run if a parent fails.</p>
<p>Is there any way to tell <strong>D</strong> to ignore failures in <strong>B</strong> and <strong>C</strong>? NB: I do not want to tell <strong>B</strong> and <strong>C</strong> to ignore failures because I do want to show them as failed in the UI in the event that they fail. I just also want to run <strong>D</strong> in that case. I currently collect the output status for both jobs, so when <strong>D</strong> runs I can tell which jobs succeeded and which failed:</p>
<pre><code class="lang-yaml">outputs:
  job_status: ${{ job.status }}
</code></pre>
<p>Alternatively, is there a way for me to tell <strong>D</strong> it should always run if <strong>A</strong> runs?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/nirvdrum">@nirvdrum</a>,</p>
<p>You can use a combined if expression in JobD. Code sample as below:</p>
<pre><code class="lang-auto">  jobD:
    needs: [jobA, jobB, jobC]                                            #depend on jobA,B,C
    if: "always()&amp;&amp;(needs.jobA.outputs.job_status=='success')"   # check jobA outputs as well.
</code></pre>
<p>the workflow will ignore jobB, jobC status and run based on jobA succeeds. Please change the jobA needs outputs according to your situation.</p>
<p>Thanks.</p>

**Comment by nirvdrum** [comment]
<p>That worked. Thanks! I tried many different variations of that <code>if</code> condition and never would have thought to combine <code>always()</code> with another condition.</p>
